services/consensusalgo/leanhelixconsensus/test/metrics_test.go

```diff
@@ -0,0 +1,127 @@
+
+		h.expectGossipSendLeanHelixMessage()
+		h.expectValidateTransactionBlock()
+		h.expectValidateResultsBlock()
+		h.expectCommitBlock()
+
+		h.dontBeFirstInCommitee()
+
+		const syncedBlockHeight = 1
+		h.handleBlockSync(ctx, syncedBlockHeight)
+
+		orderedCommittee := h.requestOrderingCommittee(ctx)
+		orderedComitteeNodeIndicies := keys.NodeAddressesForTestsToIndexes(orderedCommittee.NodeAddresses)
+
+		leaderNodeIndex := orderedComitteeNodeIndicies[0] // leader for view 0 is always the first member in the ordered committee
+
+		prevBlockProof := protocol.BlockProofReader(nil) // the previous block (from sync) had no proof
+		randomSeed := randomseed.CalculateRandomSeed(prevBlockProof.RandomSeedSignature())
+
+		const firstTermHeight = syncedBlockHeight + 1
+		const view = 0
+		blockPair := builders.BlockPair().WithHeight(firstTermHeight).WithTransactions(1).WithEmptyLeanHelixBlockProof().Build()
+
+		h.handlePreprepareMessage(ctx, blockPair, firstTermHeight, view, leaderNodeIndex)
+
+		commitMessages := []*interfaces.CommitMessage{}
+		for i := 0; i < h.networkSize(); i++ {
+			cmsg := h.handleCommitMessage(ctx, blockPair, firstTermHeight, view, randomSeed, i)
+			commitMessages = append(commitMessages, cmsg)
+		}
+
+		metrics := h.getMetrics()
+
+		// At this point the first block should be committed, lastCommitTime should update
+		require.True(t, test.Eventually(1*time.Second, func() bool {
+			return metrics.lastCommittedTime.Value() != 0
+		}), "expected lastCommittedTime metric not to update on first commit")
+
+		// timeSinceLastCommitMillis will NOT update because this is the first commit
+		require.True(t, strings.Contains(metrics.timeSinceLastCommitMillis.String(), "samples=0"), "expected lastCommittedTime to not update on first commit")
+
+		// Starting the second term
+
+		// Use commits from previous term to calculate the new random seed
+		proof := blockproof.GenerateLeanHelixBlockProof(h.keyManagerForNode(h.myNodeIndex()), commitMessages)
+		prevBlockProof = protocol.BlockProofReader(proof.Raw())
+		randomSeed = randomseed.CalculateRandomSeed(prevBlockProof.RandomSeedSignature())
+
+		const secondTermHeight = firstTermHeight + 1
+		secondBlockPair := builders.BlockPair().WithHeight(secondTermHeight).WithTransactions(1).WithEmptyLeanHelixBlockProof().Build()
+
+		h.handlePreprepareMessage(ctx, secondBlockPair, secondTermHeight, view, leaderNodeIndex)
+
+		for i := 0; i < h.networkSize(); i++ {
+			h.handleCommitMessage(ctx, secondBlockPair, secondTermHeight, view, randomSeed, i)
+		}
+
+		// A second commit should take place now and this time timeSinceLastCommitMillis should update
+
+		require.True(t, test.Eventually(1*time.Second, func() bool {
+			matched, err := regexp.MatchString("samples=[1-9]", metrics.timeSinceLastCommitMillis.String())
+			if err != nil {
+				panic(err)
+			}
+			return matched
+		}), "expected timeSinceLastCommitMillis metric to update")
+	})
```

Comment:
consider adding a third term which does everything identically except it omits the final commit message. then require that no changes to `lastCommittedTime ` are made and that `timeSinceLastCommitMillis ` increases with the clock?

being the third step it would also justify the refactor i asked for in the comment to line 100 by the so called "rule of three"

Reply:
I can also add this check in the first or second term right before emitting the final commit. So no three yet :)

btw - timeSinceLastCommit is currently only updated as a result of a commit, so it will not increase with the clock. Not sure if this was the intended behaviour..

Reply:
`timeSinceLastCommit ` - the name is misleading. since it's a histogram it means that this is the actual intended way to measure it. but then maybe a different name such as `timeDeltaC2C` or `elapsedTimeToCommit` or `timeElapsedBetweenCommits`